### PR TITLE
Iss38to40

### PR DESCRIPTION
**Pull request by bclinthall**
Resolves issues #38, #39, and #40 to implement handling of weird things that can happen.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Created by https://www.gitignore.io/api/java,gradle
+.DS_Store
```
Comment (whereswaldon) [inline .gitignore]: THANK YOU! (I hate that file)

File: src/main/java/Extractor.java [hunk 2]
```diff
@@ -21,6 +21,9 @@ public static String getProjectJSON(String sb2Path) throws IOException {
         // http://www.oracle.com/technetwork/articles/java/compress-1565076.html
         ZipFile zipFile = new ZipFile(sb2Path);
         ZipEntry entry = zipFile.getEntry("project.json");
+        if (entry == null) {
```
Comment (whereswaldon) [inline src/main/java/Extractor.java]: Did we really not check for that before...? Whoops

File: src/main/java/Sb2.java [hunk 3]
```diff
@@ -11,23 +11,40 @@
  * @author B. Clint Hall
  */
 public class Sb2 {
+    private static final String NO_JSON = "This project contains no "
```
Comment (whereswaldon) [inline src/main/java/Sb2.java]: Looks like you've found the checkstyle-approved way to declare a static final field. I'll use this for reference.

Comment (bclinthall) [inline src/main/java/Sb2.java]: @whereswaldon It seems like it wants to make sure all the static methods are before any instance methods.

**Pull request merged**